### ISSUE TITLE
Docker ignores HTTP Proxy settings completly

### ISSUE THREAD

**Issue by GitHubTestpeter** [docker/for-linux#1444]
Hi all,

first at all, I found this closed issue and I think, that I have the same problem: #1430 

I am running at work a debian 10 server with docker installed. This server has no direct internet connection. It has to use our company HTTP(S) Proxy with authentication required. I didnt changed anything in configuration files, after the server was up and running. Only updates via apt became installed.

I cant pull any image at the moment. When I try `docker run hello-world` I get this error:
`docker: Error response from daemon: Get “https://registry-1.docker.io/v2/”: net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers).`

I followed this guide, to set the proxy settings for docker: https://docs.docker.com/config/daemon/systemd/

I also have setup a lab environment at home and reproduced the error: 
I installed two Debian VMs, one as my proxy and one as the client with docker.

1. VM Proxy:
- Debian 11
- 2 Network cards (Internet, LAN)
- Squid as HTTP Proxy with Basic_auth enabled
- Bind9 DNS Forwarder

2 . VM Docker and Proxy Client
- Debian 11
- 1 Network card (LAN)
- Docker installed following the official How-To for debian

- I set up the environment variable
`export http_proxy=http://user:password@192.168.100.1:3128`

- Docker proxy configured again with this How-To: https://docs.docker.com/config/daemon/systemd/

- The file `/etc/systemd/system/docker.service.d/http-proxy.conf` contains:
```
[Service]
Environment="HTTP_PROXY=http://user:password@192.168.100.1:3128/"
```

- The output of `systemctl show --property=Environment docker` is:
`Environment=HTTP_PROXY=http://user:password@192.168.100.1:3128/`

In my opinion it should work now, to pull images, but it isnt. 
When I try `docker run hello-world`, this error occurs, same as at work.
`docker: Error response from daemon: Get "https://registry-1.docker.io/v2/": net/http: request canceled while waiting for connection (Client.Timeout exceeded while awaiting headers).`

The credentials are really user and password, so no URL-Encoding should be necessary.

docker --version
Docker version 20.10.22, build 3a2c30b

Host OS:
Debian 11 (fully updated)

Thanks in advance.

**Comment by KingCartman** [comment]
I have the same problem. 
I solved it temporarily by installing docker from the official debian repo.

What have I done:
```
apt remove docker-ce docker-ce-cli containerd.io docker-compose-plugin
apt autoremove
rm /etc/apt/sources.list.d/docker.list
apt update
apt install docker.io docker-compose
systemctl start docker
```

The working version for me is: "Docker version 20.10.5+dfsg1, build 55c4c88"

Maybe this helps to fix the issue

**Comment by KingCartman** [comment]
Okay, I fixed the problem at work too. The Account, who opened this issue is my account, I use at work. I will close it tomorrow from there.

In my lab, I forgot to set the HTTPS_PROXY variable. That causes the issue there.
The content from `/etc/systemd/system/docker.service.d/http-proxy.conf` has to be:
```
[Service]
Environment="HTTP_PROXY=http://user:password@192.168.100.1:3128"
Environment="HTTPS_PROXY=http://user:password@192.168.100.1:3128"
```

What was the Problem at work? A DNS problem... I have to solve that, until that I entered the IP-Address there instead of the FQDN.

It has nothing to do with the docker version. I am sorry.

But maybe this helps someone in the future.
